input/postgres/backends.go

```diff
@@ -63,7 +77,8 @@ func GetBackends(logger *util.Logger, db *sql.DB, postgresVersion state.Postgres
 			&row.RoleOid, &row.RoleName, &row.Pid, &row.ApplicationName, &row.ClientAddr,
 			&row.ClientPort, &row.BackendStart, &row.XactStart, &row.QueryStart,
 			&row.StateChange, &row.Waiting, &row.BackendXid, &row.BackendXmin,
-			&row.WaitEventType, &row.WaitEvent, &row.BackendType, &row.State, &row.Query)
+			&row.WaitEventType, &row.WaitEvent, &row.BackendType, pq.Array(&row.BlockedByPids),
+			&row.State, &row.Query)
```

Comment:
It looks like [pg_blocking_pids](https://github.com/postgres/postgres/blob/9fd45870c1436b477264c0c82eb195df52bc0919/src/backend/utils/adt/lockfuncs.c#L429-L437) includes duplicates for parallel queries. Maybe we should remove those duplicates? (In Go since Postgres doesn't seem to have a good function to dedup an array)

Reply:
thanks for sharing that! given that the duplicates count won't big (only for parallel queries), also we are going to remove dups in the backend/frontend part when presenting, I think we don't need to focus on removing duplicates here, so I'm going to leave it as is 👍